### PR TITLE
INVGOV-10702 Add /job_roles/list

### PR DESCRIPTION
**Pull request by sam-ludlow**
https://io1dev.atlassian.net/browse/INVGOV-10702

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -2125,6 +2125,55 @@ paths:
             schema:
               $ref: '#/components/schemas/job-role-schema-create'
         required: true
+  '/job_roles/list':
+    get:
+      tags:
+        - 'job roles'
+      description: 'Search for job roles. An empty array is returned if no job roles are found.'
+      parameters:
+        - name: name
```
Comment (ParryLuke) [inline api.swagger.yaml]: Should be title

File: api.swagger.yaml [hunk 2]
```diff
@@ -2125,6 +2125,55 @@ paths:
...
+          description: 'Filter by name. Matches on whole words.'
```
Comment (ParryLuke) [inline api.swagger.yaml]: title

Comment (sam-ludlow) [reply api.swagger.yaml]: done

File: api.swagger.yaml (hunk 1 above)
Comment (sam-ludlow) [reply api.swagger.yaml]: done

**Comment by sam-ludlow** [comment]
Redoing PR due to merge conflicts.

**Pull request closed without merging**